pkg/mcp/rate/ratelimiter.go

```diff
@@ -21,65 +21,35 @@ import (
 	"golang.org/x/time/rate"
 )
 
-// RateLimit is partially representing standard lib's rate limiter
-type RateLimit interface {
+// Limit is partially representing standard lib's rate limiter
+type Limit interface {
 	Wait(ctx context.Context) (err error)
 }
 
-var _ RateLimit = &rate.Limiter{}
-
 // ConnectionRateLimit is an interface for creating per-connection rate limiters
 type ConnectionRateLimit interface {
-	Create() RateLimit
+	Create() Limit
```

Comment:
Can we change the name of `ConnectionRateLimit` to match its intent? Everytime I see this now I think it's purpose is to rate limit connections =/ Maybe something like RateLimitFactory. cc @ozevren for naming.